service-base/src/main/java/org/eclipse/hono/service/credentials/validators/AbstractSecretsValidator.java

```diff
@@ -0,0 +1,117 @@
+ * credential secrets.
+ * <p>
+ * The class implements the validation steps that are common to all types of credentials. This e.g. includes the enabled
+ * flag of credential entries, the iteration over several valid credential entries (until one is successfully validated
+ * or none is left), etc.
+ * <p>
+ * The detailed algorithm to validate a single credential entry is delegated to the implementing subclass and so supports
+ * a specified small implementation class per credentials type..
+ *
+ * @param <T> The type of what has to be validated (called item below): this can be String in case of password validation, a certificate
+ *           class in case of a client certificate, etc.
+ */
+public abstract class AbstractSecretsValidator<T> implements SecretsValidator<T> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(AbstractSecretsValidator.class);
+
+    /**
+     * Get the type of credentials secrets this validator is responsible for.
+     * <p>This can be freely defined, but there are some predefined types in the
+     * <a href="https://www.eclipse.org/hono/api/Credentials-API/">Credentials API</a>.
+     *
+     * @return The type of credentials secrets.
+     */
+    @Override
+    public abstract String getSecretsType();
+
+    /**
+     * Validate an instance of T (e.g. a password) against a single credentials secret (as JsonObject).
+     * <p>
+     * Subclasses need to implement their specified algorithm for validation in this method.
+     *
+     * @param itemToValidate The item to validate.
+     * @param aSecret The secret record as JsonObject (as returned by the <a href="https://www.eclipse.org/hono/api/Credentials-API/">Credentials API</a>.
+     * @return The result of the validation as boolean.
+     */
+    protected abstract boolean validateSingleSecret(final T itemToValidate, final Map<String, String> aSecret);
+
+    /**
+     * Validate  an instance of T (e.g. a password) against credentials secrets (as JsonObject as defined in the
+     * <a href="https://www.eclipse.org/hono/api/Credentials-API/">Credentials API</a>).
+     * <p>
+     * The payload from the get operation of the <a href="https://www.eclipse.org/hono/api/Credentials-API/">Credentials API</a>
+     * is parsed and splitted into single secrets entries in this method. The single entries are then delegated to the
+     * {@link #validateSingleSecret} method of the implementing subclass where the detailed
+     * validation is processed. If one entry was successfully validated, the validation is considered
+     * successful and the result is completed. If no secret could be validated, the validation fails and thus the result
+     * is set to failed.
+     *
+     * @param credentialsObject The credentials that were returned from the credentials get operation.
+     * @param authenticationObject The object to authenticate.
```

Comment:
au*authenticationObject* is the *secret* to verify, right?
If so, can't we just call it *secret*? The term *authenticationObject* seems rather unspecific and arbitrary